### PR TITLE
add code of conduct

### PR DESCRIPTION
**Pull request by karenzshea**
# Issue

Closes https://github.com/Project-OSRM/osrm-backend/issues/4168

## Tasklist
 - [x] update relevant [docs](CONTRIBUTING.md)
 - [ ] review
 - [ ] adjust for comments

### CODE REVIEW

**Comment by karenzshea** [comment]
cc @Project-OSRM/team-osrm

File: CODE-OF-CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Code of conduct
```
Comment (daniel-j-h) [inline CODE-OF-CONDUCT.md]: Does the file have the be named `CODE_OF_CONDUCT` (with underscores) for Github to recognize it? Will this file then show up for issues and prs?

https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/

File: CODE-OF-CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+Everyone is invited to participate in Mapbox’s open source projects and public discussions: we want to create a welcoming and friendly environment. Harassment of participants or other unethical and unprofessional behavior will not be tolerated in our spaces. The [Contributor Covenant](http://contributor-covenant.org) applies to all projects under the Mapbox organization and we ask that you please read [the full text](http://contributor-covenant.org/version/1/2/0/).
```
Comment (TheMarex) [inline CODE-OF-CONDUCT.md]: We should replace `Mapbox` with `Project OSRM`.

File: CODE-OF-CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+You can learn more about our open source philosophy on [mapbox.com](https://www.mapbox.com/about/open/).
```
Comment (TheMarex) [inline CODE-OF-CONDUCT.md]: We can add a link to the Mapbox open source policy as reading material, but I think the COC might be the wrong place.

File: CODE-OF-CONDUCT.md (hunk 1 above)
Comment (karenzshea) [reply CODE-OF-CONDUCT.md]: Hm, I just did what https://raw.githubusercontent.com/mapbox/mapbox-gl-native/master/CODE-OF-CONDUCT.md did

**Pull request merged**